src/client/driver/driver.js

```diff
@@ -426,6 +443,105 @@ export default class Driver extends serviceUtils.EventEmitter {
         childIframeDriverLink.sendConfirmationMessage(id);
     }
 
+    _getTargetWindowFoundResult (msg) {
+        const result = {
+            success: true
+        };
+
+        if (this.childWindowDriverLinks.length && msg.cmd === MULTIPLE_WINDOW_API_COMMAND.close) {
+            result.success = false;
+            result.errCode = TEST_RUN_ERRORS.cannotCloseWindowWithChildrenError;
+        }
+
+        return Promise.resolve(result);
+    }
+
+    _getTargetWindowNotFoundResult () {
+        return Promise.resolve({
+            success: false,
+            errCode: TEST_RUN_ERRORS.targetWindowNotFoundError
+        });
+    }
+
+    _getChildWindowValidateResult (arr) {
+        const success = arr.find(item => item.result.success);
+
+        if (success)
+            return success.result;
+
+        let errItem = arr.find(item => item.result.errCode === TEST_RUN_ERRORS.cannotCloseWindowWithChildrenError);
+
+        if (!errItem)
+            errItem = arr.find(item => !!item.result.errCode);
+
+        return errItem ? { errCode: errItem.result.errCode } : void 0;
+    }
+
+    _handleValidateWindowExists (msg, wnd) {
+        let promiseResult = null;
+
+        if (msg.windowId === this.windowId)
+            promiseResult = this._getTargetWindowFoundResult(msg);
+
+        else if (!this.childWindowDriverLinks.length)
+            promiseResult = this._getTargetWindowNotFoundResult();
+
+        else {
+            promiseResult = Promise.all(this.childWindowDriverLinks.map(childWindowDriverLink => {
+                return childWindowDriverLink.searchChildWindows(msg, WindowValidationMessage);
+            }))
+                .then(arr => {
+                    return this._getChildWindowValidateResult(arr);
+                });
+        }
+
+        return promiseResult
+            .then(result => {
+                sendConfirmationMessage({
+                    requestMsgId: msg.id,
+                    window:       wnd,
+                    result
+                });
+            });
+    }
+
+    _handleExecuteWindowApiCommand (msg, wnd) {
```

Comment:
I propose to split this command to `_handleWindowCloseCommand` and `_handleSwitchToWindowCommand`.

Reply:
I splitted this command into parts. However, I was not able to merge `validation` and `execute command` steps. So right now I do tree search twice: for validation and for execute.
I do this, because of difficulties in close command with `cannotCloseWindowWithChildren` part.

Reply:
The problem is when I close window, I need to close child window from parent driver using childDriverWindowLink.driverWindow - because in parent driver I need to handle the moment when child window is closed.
When I validate window I need to check windowId not form childDriverWindowLink, but from the child driver itself, because I need to check that childDriver does not contain other children.